torch/optim/optimizer.py

```diff
@@ -391,10 +391,39 @@ def state_dict(self):
 
         It contains two entries:
 
-        * state - a dict holding current optimization state. Its content
-            differs between optimizer classes.
-        * param_groups - a list containing all parameter groups where each
-            parameter group is a dict
+        * ``state`` - a Dict holding current optimization state. Its content
```

Comment:
Do we actually match the id order with the params in the opt we load this into? Or the order of the "state" dict ?

Reply:
So it assumes that the two are equivalent and matches the param_groups containing params in the opt to the param ids of the state. For example, if our saved state_dict had:

```
'param_groups': [{ ...,  'params': [0, 1]},  { ..., 'params': [2, 3]}]
```

and our param_groups in the opt we're loading into has:
```
'param_groups': [{ ...,  'params': [tensor_a, tensor_b]},  { ..., 'params': [tensor_c, tensor_d]}]
```

It would just load the state of the 0 id'd param into tensor_a, the 1 id'd param state into tensor_b, and so forth. Does this answer your question?

Reply:
My question is really that this doc should explain how we read this state dict if it defines what it contains.
Preferably a single sentence version of this? :D